setools/policyrep/boolcond.pxi

```diff
@@ -78,50 +78,77 @@ cdef class Conditional(PolicyObject):
         return other in self.booleans
 
     def __str__(self):
-        # sepol representation is in postfix notation.  This code
```

Comment:
Here and in other places, where there isn't a statement, it should be handled using __format__(), also with a default of kernel output. Please support `kernel` and `cil` as format options. See `setools/dta.py: DomainTransition` for an example.

Reply:
Thinking more about your example output, this is not meant to output a full conditional block, so it should not be rendering a full booleanif/(true/(false block in the cil output.

Reply:
> Here and in other places, where there isn't a statement, it should be handled using **format**(), also with a default of kernel output. Please support `kernel` and `cil` as format options. See `setools/dta.py: DomainTransition` for an example.

That looks nice, thanks for the pointer.

Reply:
> Thinking more about your example output, this is not meant to output a full conditional block, so it should not be rendering a full booleanif/(true/(false block in the cil output.

The purpose and use case of the CIL output flag (which I should have stated in the patch description) is that when the output is set to CIL, I can copy and use the text mechanically without modification in other CIL policies. It's also possible to compare the original CIL policy and the output of  `sesearch`. If you invent a new syntax which is not CIL, the output would need to be converted to CIL somehow (perhaps with another tool) and the flag would become much less useful. The same problem exists in the kernel syntax version, the output is not valid KPL syntax either.

Reply:
> The same problem exists in the kernel syntax version, the output is not valid KPL syntax either.

This is the intent: to succinctly return the results. Not the intent: copy the result verbatim to a policy source file. If I have 50 conditional results it would be a mess. The kernel language was chosen for rendering because SETools predates CIL. SETools did not change to CIL rendering because it is an intermediate language that most don't even see.

Reply:
Since CIL is an intermediate language, I don't feel the need to extensively support as you describe.